packages/workspace/src/workspace/reference_indexes.ts

```diff
@@ -113,30 +123,35 @@ const getReferenceTargetIndexUpdates = (
       ...Object.values(allFields)
         .map(field => ({
           key: field.elemID.getFullName(),
-          value: baseIdToReferences[field.elemID.getFullName()] ?? [],
+          value: new collections.treeMap.TreeMap<ElemID>(
+            (baseIdToReferences[field.elemID.getFullName()] ?? []).map(ref =>
+              [ref.referenceSource.getFullName(), [ref.referenceTarget]])
+          ),
```

Comment:
maybe worth extracting this logic to a function just to make the code more readable.
this is borderline in terms of pure readability, but it is also duplicated so there is more motivation to extract to a simple function, I think this passes the threshold and will look better if we extract to a function.


```suggestion
          value: createReferenceTree(baseIdToReferences[field.elemID.getFullName()] ?? []),
```